folly/memory/UninitializedMemoryHacks.h

```diff
@@ -110,6 +130,11 @@ inline void resizeWithoutInitialization(
       s.reserve(n);
     }
     detail::unsafeStringSetLargerSize(s, n);
+#if defined(_MSC_VER) && _MSC_VER >= 1938 && defined(_INSERT_STRING_ANNOTATION)
+    if (_Asan_string_should_annotate && s.capacity() > std::string().capacity())
+      __sanitizer_annotate_contiguous_container(&s[0], &s[0] + s.capacity() + 1,
+          &s[0] + n0 + 1, &s[0] + n + 1);
+#endif
```

Comment:
This looks like library-specific implementation that must vary by library and library version. So it should be in the relevant per-library implementations, below. For clarity, see where the existing calls to `__sanitizer_annotate_contiguous_container` are located.